parakeet_mlx/alignment.py

```diff
@@ -20,12 +21,18 @@ class AlignedSentence:
     start: float = 0.0  # temporary
     end: float = 0.0  # temporary
     duration: float = 0.0  # temporary
+    confidence: float = 1.0  # aggregate confidence score
 
     def __post_init__(self) -> None:
         self.tokens = list(sorted(self.tokens, key=lambda x: x.start))
         self.start = self.tokens[0].start
         self.end = self.tokens[-1].end
         self.duration = self.end - self.start
+        # Compute geometric mean of token confidences
+        if self.tokens:
+            import mlx.core as mx
+            confidences = mx.array([t.confidence for t in self.tokens])
+            self.confidence = float(mx.exp(mx.mean(mx.log(confidences + 1e-10))))
```

Comment:
We can assume that tokens are not empty list. (Already assuming in line 27!) Also, MLX can be pretty slow at this immediate calculation since it [builds graph](https://gist.github.com/awni/4beb1f7dfefc6f9426f3a7deee74af50) first. I think we can use NumPy here! Also I think import can be placed on top of the file.

```suggestion
        confidences = np.array([t.confidence for t in self.tokens])
        self.confidence = float(np.exp(np.mean(np.log(confidences + 1e-10))))
```